modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingApis.ts

```diff
@@ -0,0 +1,68 @@
+import { Optional, Optionals } from '@ephox/katamari';
+import { Attribute } from '@ephox/sugar';
+import * as Behaviour from '../../api/behaviour/Behaviour';
+import { Keying } from '../../api/behaviour/Keying';
+import { Focusing } from '../../api/behaviour/Focusing';
+import { Replacing } from '../../api/behaviour/Replacing';
+import { AlloySpec } from '../../api/component/SpecTypes';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import * as GuiFactory from '../../api/component/GuiFactory';
+import { BlockingConfig } from './BlockingSchema';
+import { BlockingState } from './BlockingState';
+
+export type BlockFn = (component: AlloyComponent, config: BlockingConfig, state: BlockingState, getBusySpec?: (behaviour: Behaviour.AlloyBehaviourRecord) => AlloySpec) => void;
```

Comment:
You mentioned you'd swapped `getBusySpec` back so the first argument is a component, but as far as I can see that hasn't been done and this is still an API breaking change.

Reply:
I swapped it for the ModalDialog, but the behaviour itself can't have any blocking API changes because it didn't exist before

Reply:
https://github.com/tinymce/tinymce/pull/6196/files#diff-f752f5911597c41388a68d8de05593894b53d172f2781c5fff531a7acf0a543c says otherwise?

Reply:
Also, why don't we want to pass the component (root or current) through in `getBusySpec`?

Reply:
I would agree with that as a general rule. Because in alloy, component references can sometimes be hard to come by, you nearly always regret not providing a component to a callback function.

Reply:
Added this parameter